v-next/hardhat-keystore/src/internal/ui/direct-user-interruption-manager.ts

```diff
@@ -0,0 +1,87 @@
+import type { UserInterruptionManager } from "@ignored/hardhat-vnext/types/user-interruptions";
+
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+
+/**
+ * A class that handles user interruptions directly on the console
+ * and explicitly does NOT go through the user interruptions
+ * system of the Hook system.
+ */
+export class DirectUserInterruptionManager implements UserInterruptionManager {
```

Comment:
I don't get this. I think it shouldn't exist, it conflicts with the user interruptions system. It's meant to be used "as is" by a plugin like this, so that others can override its behavior.